pkg/secrethub/org.go

```diff
@@ -73,3 +76,41 @@ func (s orgService) Members() OrgMemberService {
 func (s orgService) ListMine() ([]*api.Org, error) {
 	return s.client.httpClient.ListMyOrgs()
 }
+
+// Iterator returns an iterator that lists all organizations of the current users.
+func (s orgService) Iterator() OrgIterator {
+	data, err := s.ListMine()
+	return &orgIterator{
+		index: 0,
+		data:  data,
+		err:   err,
+	}
+}
+
+// RepoIteratorParams defines parameters used when listing repos.
+type OrgIteratorParams struct{}
+
+// RepoIterator iterates over repositories.
+type OrgIterator interface {
+	Next() (api.Org, error)
+}
+
+type orgIterator struct {
```

Comment:
Nice work Márton! Just had a quick read through and one question popped in my mind:

Are these iterators implemented in a thread-safe / goroutine safe way? And more importantly, should they?

If you have questions on this, discuss it with @SimonBarendse I know he has thoughts about this 😄

Reply:
Good question(s)! I will discuss this with @SimonBarendse on Monday.

Reply:
Yes, would be nice to make it safe for concurrent access. Especially since this is also done for the audit event iterator, so to be consistent it would be good to also make these iterators safe for concurrent access.

Have a look here: https://github.com/secrethub/secrethub-go/blob/v0.25.0/pkg/secrethub/iterator/iterator.go#L40 how this is done for the audit iterator.

Another possibility is to already adopt the iterator/paginator pattern and use [`iterator.Iterator`](https://github.com/secrethub/secrethub-go/blob/v0.25.0/pkg/secrethub/iterator/iterator.go#L22), which already provides the safety for concurrent access. The paginator could return a single page for now.
This will also make it easier to switch to paginated pages later, as we'll only have to swap out the paginator.